### DISCUSSION TITLE
Microstructure segmentation

### DISCUSSION THREAD

**Discussion by makeabhishek** [mtex-toolbox/mtex#1015] [Ask Anything]
Hi there,
I would like to calculate the average microtextured regions (MTRs) or macrozones size in the horizontal and vertical direction using a linear intercept method as shown below figure. Is there any method to convert the orientation map to a binary image with boundary regions?

![EBSD_segmentation](https://user-images.githubusercontent.com/47937684/117423966-dbd13780-af53-11eb-9d1c-d64674c1e638.JPG)

**Comment by kilir** [comment]
Hi,
you can use `grainBoundary/intersect` on the boundary segments to get line intercepts. In case you do not only want to consider fully closed domains but all boundary segments fulfilling a certain property (e.g. threshold angle) also consider `.innerBoundary` in addition to `.boundary`. See here https://mtex-toolbox.github.io/BoundaryIntersections.html

Hope that helps.

Cheers,
Rüdiger

**Reply by makeabhishek** [reply]
Thanks, Rüdiger,
I tried that but cannot separate the grain boundaries from the microtextured regions. The boundary image looks like that.
![image](https://user-images.githubusercontent.com/47937684/117433268-e395d980-af5d-11eb-83bb-cd04283c7559.png)

Regards
Abhishek

**Reply by kilir** [reply]
Hi,
it looks like your segmentation does not apply very well to the type of data, i.e. very noisy data, inappropriate type of segmentation etc. What segmentation do you use, what data is it, what is your step size with respect to what information you have about the data? 
Cheers,
Rüdiger

**Reply by makeabhishek** [reply]
Hi,
The scanning step considered in EBSD scanning is 10 micrometre. A large area scan was performed for a Ti64 dual phase alloy. My first post shows the approx representation of microstructure. I want to achieve a similar segmentation to capture the elongated regions, line intercept method can be used to quantify the size of the regions.
Reference: Pilchak, Adam L., Jia Li, and Stanislav I. Rokhlin. "Quantitative comparison of microtexture in near-alpha titanium measured by ultrasonic scattering and electron backscatter diffraction." Metallurgical and Materials Transactions A 45.10 (2014): 4679-4697.
Here is the single-phase orientation plot:
![image](https://user-images.githubusercontent.com/47937684/117455163-80b23b80-af79-11eb-95dc-2276daf35045.png)

The grain boundary method I used in Mtex is
```
fileName = [EBSD_filePath filesep EBSD_filename];
ebsd = EBSD.load(fileName);
mineral_name = ebsd.CSList{1,2}.mineral;
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',15*degree);
grains = grains.smooth(5);
plot(grains(mineral_name),grains(mineral_name).meanOrientation)

```
I also tried to remove grains but that does not work well
```
% remove two pixel grains
ebsd(grains(grains.grainSize<=2)) = [];
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',5*degree,'removeQuadruplePoints');

% smooth them
grains = grains.smooth(5);

% visualize the grains % plot(grains,grains.meanOrientation)
plot(grains(mineral_name),grains(mineral_name).meanOrientation)
```

**Reply by kilir** [reply]
What do you exactly mean by "does not work well"? Have you tried calcGrains on the entire dataset instead of just 'indexed'? Also, did you try to use a smoothing filter? Alternatively, you could try one of the clustering methods (e.g.) instead of a purely angle-threshold based method.
Cheers,
Rüdiger

**Comment by lfgmorales** [comment]
Hi

Your map is very noisy, probably due to bad sample preparation, you barely see these elongated red grains, but nothing in between. You have to repolish your sample appropriately, redo the ebsd mapping and then processed the data using the script you have. Any attempt to clean up this will create unwanted artifacts

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 7 May 2021, at 15:18, Abhishek saini ***@***.***> wrote:

﻿

Hi,
The scanning step considered in EBSD scanning is 10 micrometre. A large area scan was performed for a Ti64 dual phase alloy. My first post shows the approx representation of microstructure. I want to achieve a similar segmentation to capture the elongated regions, line intercept method can be used to quantify the size of the regions.
Reference: Pilchak, Adam L., Jia Li, and Stanislav I. Rokhlin. "Quantitative comparison of microtexture in near-alpha titanium measured by ultrasonic scattering and electron backscatter diffraction." Metallurgical and Materials Transactions A 45.10 (2014): 4679-4697.
Here is the single-phase orientation plot:
[image]<https://user-images.githubusercontent.com/47937684/117455163-80b23b80-af79-11eb-95dc-2276daf35045.png>

The grain boundary method I used in Mtex is

fileName = [EBSD_filePath filesep EBSD_filename];
ebsd = EBSD.load(fileName);
mineral_name = ebsd.CSList{1,2}.mineral;
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',15*degree);
grains = grains.smooth(5);
plot(grains(mineral_name),grains(mineral_name).meanOrientation)



I also tried to remove grains but that does not work well

% remove two pixel grains
ebsd(grains(grains.grainSize<=2)) = [];
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',5*degree,'removeQuadruplePoints');

% smooth them
grains = grains.smooth(5);

% visualize the grains % plot(grains,grains.meanOrientation)
plot(grains(mineral_name),grains(mineral_name).meanOrientation)


—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-709258>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INZ5D6LOOSTIHDPPQOTTMPSDXANCNFSM44JJ464Q>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-709258", "url": "https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-709258", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by makeabhishek** [reply]
I think the EBSD map is ok. Zero solutions are not more than 10 %
![image](https://user-images.githubusercontent.com/47937684/117464036-d93a0680-af82-11eb-8d2e-01873fd8c3cb.png)

**Reply by kilir** [reply]
It looks like your have many single point solutions, right? How can you be confident in a single point?

**Reply by makeabhishek** [reply]
this is a large scan map with a scanning step of 10 um. I am interested in the microtexture region (like the rolled elongated structure.) the sample is a forged sample with an average grain size of 6 um.

**Comment by makeabhishek** [comment]
What do you mean by single point?

Best Regards
Abhishek

**Comment by lfgmorales** [comment]
If you follow a single line of pixels on your map, you see that in many parts you have one pixel of color x, followed right away by one with color y, and the next with color z. There is a very good chance that those are badly indexed points, because many factors, one of them being sample preparation. And if you say you have an alloy with average grain size of 6 microns, but do a map with stepsize of 10, the maximum you can get after removing the incorrect measurements is a bulk texture, but no ingormation related to grains

Best

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 7 May 2021, at 18:26, Abhishek saini ***@***.***> wrote:

﻿

What do you mean by single point?

Best Regards
Abhishek

> On 7 May 2021, at 11:44 PM, Rüdiger Kilian ***@***.***> wrote:
>
> ﻿
> It looks like your have many single point solutions, right? How can you be confident in a single point?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-710203>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN5BGTIHPRVCG66FTDTTMQICBANCNFSM44JJ464Q>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-710203", "url": "https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-710203", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by makeabhishek** [comment]
Yes, exactly I want bulk texture information rather then grain info.
As you could see that multiple groans combine to form a large macrottexture regions. I am interested to measure those. 

Best Regards
Abhishek

**Comment by lfgmorales** [comment]
Then first go here (remove inaccurate measurements)

https://mtex-toolbox.github.io/EBSDSelect.html

Then here

https://mtex-toolbox.github.io/EBSDAdvancedMaps.html

Be aware that, because your map is very noisy,  many of those pixels will disappear when you clean up the map and your „bulk texture“ will  represent only the the pixels that remain in the map, and certainly not the bulk texture of your sample. Like I said before, you may want to try to repolish-remeasure your sample for better results

Cheers

Lulz



Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 8 May 2021, at 06:32, Abhishek saini ***@***.***> wrote:

﻿

Yes, exactly I want bulk texture information rather then grain info.
As you could see that multiple groans combine to form a large macrottexture regions. I am interested to measure those.

Best Regards
Abhishek

> On 8 May 2021, at 11:55 AM, Luiz Morales ***@***.***> wrote:
>
> ﻿
> If you follow a single line of pixels on your map, you see that in many parts you have one pixel of color x, followed right away by one with color y, and the next with color z. There is a very good chance that those are badly indexed points, because many factors, one of them being sample preparation. And if you say you have an alloy with average grain size of 6 microns, but do a map with stepsize of 10, the maximum you can get after removing the incorrect measurements is a bulk texture, but no ingormation related to grains
>
> Best
>
> Luiz
>
> Luiz F. G. Morales
> ScopeM - ETH Zürich
> Sent from iPhone
>
> On 7 May 2021, at 18:26, Abhishek saini ***@***.***> wrote:
>
> ﻿
>
> What do you mean by single point?
>
> Best Regards
> Abhishek
>
> > On 7 May 2021, at 11:44 PM, Rüdiger Kilian ***@***.***> wrote:
> >
> > ﻿
> > It looks like your have many single point solutions, right? How can you be confident in a single point?
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-710203>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN5BGTIHPRVCG66FTDTTMQICBANCNFSM44JJ464Q>.
>
> [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-710203", "url": "https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-710203", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-712365>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4Y3IKZS5LJZAQ75U3TMS5FXANCNFSM44JJ464Q>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-712365", "url": "https://github.com/mtex-toolbox/mtex/discussions/1015#discussioncomment-712365", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]
